src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -54,6 +59,33 @@ public OverseasEntitiesService(OverseasEntitySubmissionsRepository overseasEntit
         this.dateTimeNowSupplier = dateTimeNowSupplier;
     }
 
+    public SubmissionType getSubmissionType(String requestId, String overseasEntityId) throws SubmissionNotFoundException {
+        if (!isROEUpdateEnabled) {
+            return SubmissionType.REGISTRATION;
+        }
+        Optional<OverseasEntitySubmissionDto> submissionOpt = getOverseasEntitySubmission(
+                overseasEntityId);
+        if (submissionOpt.isEmpty()) {
+            var errorMessage = "Can not determine submission type";
+            Map<String, Object> logMap = new HashMap<>();
+            logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
+            var error = new SubmissionNotFoundException(errorMessage);
+            ApiLogger.errorContext(requestId, errorMessage, error, logMap);
+            throw error;
+        }
+
+        String registrationNumber = submissionOpt.get().getEntity().getRegistrationNumber();
```

Comment:
We can't use the 'registration number' field on 'entity' to determine if this is an UPDATE or CREATE as this is used by the registration journey to record the registration number in the country they were formed in. See screenshot.

Update records should potentially be saved in a separate Mongo collection to the registration records, so this requires a higher-level discussion, I think.

<img width="853" alt="Screenshot 2023-01-16 at 17 35 22" src="https://user-images.githubusercontent.com/21172560/212880302-1ce00580-676f-4263-bcf5-5e4c5c95d2fd.png">

Reply:
This was discussed in our ODP https://companieshouse.atlassian.net/wiki/spaces/ROE2/pages/4014178319/ODP-ROE-UPDATE-003+Payment+for+Update (with CH Architects) and the decision was to not have separate API endpoints. To use the oe_number and ( for remove) cessation date to differentiate the submission type. and But I now see we that I've missunderstood the registration number and we need to add a new field to the submission. But let us discuss.

Reply:
Now changed to use oe_number which will have to go into the AppData model in Web.

Reply:
👍 will resolve this, although this approach is still under review.